lib/game.js

```diff
@@ -0,0 +1,103 @@
+class Game {
+  constructor(
+    grid = {
+      1: 1,
+      2: 2,
+      3: 3,
+      4: 4,
+      5: 5,
+      6: 6,
+      7: 7,
+      8: 8,
+      9: 9,
+    },
+    players = {
+      O: null,
+      X: null,
+    },
+    nextPlayer = null,
+    choices = [],
+    winner = false,
+    mode = null
+  ) {
+    this.grid = grid;
+    this.players = players;
+    this.nextPlayer = nextPlayer;
+    this.choices = choices;
+    this.winner = winner;
+    this.mode = mode;
+  }
+  getPlayerName(player) {
+    return this.players[player].name;
+  }
+  getNextPlayerName() {
+    return this.players[this.nextPlayer].name;
+  }
+  randomTeams() {
+    const player1 = Math.random() < 0.5 ? "O" : "X";
+    const player2 = player1 === "O" ? "X" : "O";
+    return [player1, player2];
+  }
+  setMode(mode) {
+    this.mode = mode;
+  }
+  // FIXUP - added a test parameter to facilitate testing in short term
+  setPlayOrder(test) {
```

Comment:
In js, historically every parameter is optional, but it's useful to flag this to the reader. In other languages, optional parameters *must* use functionName(parameterName = default) syntax. I would recommend using that syntax here.

I think I would write this as `setPlayOrder(team = null) {` and then `if (!team) { team = Math.random() < 0.5 ? "X" : "O" }` or something.